bin/java

```diff
@@ -111,6 +111,31 @@ Heroku"
   fi
 }
 
+download_binary() {
+  local jdkUrl="${1:?}"
+  local jdkTarball="${2:?}"
+  local curDir=${JVM_COMMON_DIR:-$(cd $(dirname ${BASH_SOURCE[0]}) && cd .. && pwd )}
+
+  curl --retry 3 --silent --show-error --location "${jdkUrl}" --output "${jdkTarball}"
+
+  if [ "${HEROKU_GPG_VALIDATION:-0}" != "1" ]; then
+    _jvm_mcount "gpg.verify.skip"
+  else
+    curl --retry 3 --silent --show-error --location "${jdkUrl}.gpg" --output "${jdkTarball}.gpg"
+
+    gpg --no-tty --batch --import "${curDir}/.gnupg/lang-jvm.asc" > /dev/null 2>&1
+
```

Comment:
@dzuelke Should I add a step here to handle revocations?

Reply:
A revocation for an existing key (along with a new key) would also be contained in `lang-jvm.asc`, so I think this is fine for now.

Reply:
but what if someone is using an old version of the buildpack, which doesn't have a revocation? Don't we want to put the revocations on S3 for that reason?